### ISSUE TITLE
False Positive "unresolved-import"

### ISSUE THREAD

**Issue by hwsimmons17** [rust-analyzer/rust-analyzer#11313]
I am getting a false positive error in "unresolved-import" when I try to import a struct generated by a proc-macro in a separate crate.

Is there a way to override the error and quiet it, or maybe it could just be fixed. In the source code, it is clear the devs are aware of the issue

``// This currently results in false positives in the following cases:
    // - `cfg_if!`-generated code in libstd (we don't load the sysroot correctly)
    // - `core::arch` (we don't handle `#[path = "../<path>"]` correctly)
    // - proc macros and/or proc macro generated code
``

![Screen Shot 2022-01-18 at 10 05 03 AM](https://user-images.githubusercontent.com/63158553/149984225-d58f156f-6461-4bb9-ae91-4dc8c9d1c014.png)

**Comment by jonas-schievink** [comment]
You can turn off the diagnostic as described here: https://rust-analyzer.github.io/manual.html#diagnostics

This should work fine though. Does the proc macro get expanded (there should be a diagnostic if not, and you can use the "Expand macro recursively" command to check)?

**Comment by kturney** [comment]
FWIW, I was getting this same error while using the 1.51 toolchain (set via rust-toolchain.toml).

Updating to latest `stable` toolchain resolved the issue.

I convinced rust-analyzer to use stable without having to update rust-toolchain.toml by using the following vs code setting:

```json
{
  "rust-analyzer.server.extraEnv": {
    "RUSTUP_TOOLCHAIN": "stable"
  }
}
```

Additionally, it appears that [only the latest stable toolchain is officially supported by rust-analyzer](https://github.com/rust-analyzer/rust-analyzer/issues/11226).
